File: hw/vargas/u2/hw19Inheritance/python/model/Location.py
class Location:
    def __init__(self, x_coordinate, y_coordinate):
        self._x_coordinate = x_coordinate
        self._y_coordinate = y_coordinate

    def __str__(self):
        return f"Location{{xCoordinate={self._x_coordinate}, yCoordinate={self._y_coordinate}}}"

    def get_x_coordinate(self):
        return self._x_coordinate

    def set_x_coordinate(self, x_coordinate):
        self._x_coordinate = x_coordinate

    def get_y_coordinate(self):
        return self._y_coordinate

    def set_y_coordinate(self, y_coordinate):
        self._y_coordinate = y_coordinate

File: hw/bonilla/u2/hw19Inheritance/FarmSystemJavaScript/model/Location.js
class Location {
    constructor(xCoordinate, yCoordinate) {
        if (new.target === Location) {
            throw new Error("Location is an abstract class and cannot be instantiated directly.");
        }

        this.xCoordinate = xCoordinate;
        this.yCoordinate = yCoordinate;
    }

    toString() {
        return `{
        "xCoordinate": ${this.xCoordinate},
        "yCoordinate": ${this.yCoordinate}
}`;
    }

    getxCoordinate() {
        return this.xCoordinate;
    }

    setxCoordinate(xCoordinate) {
        this.xCoordinate = xCoordinate;
    }

    getyCoordinate() {
        return this.yCoordinate;
    }

    setyCoordinate(yCoordinate) {
        this.yCoordinate = yCoordinate;
    }
}

module.exports = Location;
